go/ql/src/experimental/CWE-770/DenialOfService.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @name Denial Of Service
+ * @description slices created with the built-in make function from user-controlled sources using a
+ * maliciously large value possibly leading to a denial of service.
+ * @kind path-problem
+ * @problem.severity high
+ * @security-severity 9
+ * @id go/denial-of-service
+ * @tags security
+ *       experimental
+ *       external/cwe/cwe-770
+ */
+
+import go
+
+class BuiltInMake extends DataFlow::Node {
+  BuiltInMake() { this = Builtin::make().getACall().getArgument(0) }
+}
+
+/**
+ * Holds if `g` is a barrier-guard which checks `e` is nonzero on `branch`.
+ */
+predicate denialOfServiceSanitizerGuard(DataFlow::Node g, Expr e, boolean branch) {
+  exists(DataFlow::Node lesser |
+    e = lesser.asExpr() and
+    g.(DataFlow::RelationalComparisonNode).leq(branch, lesser, _, _)
+  )
+  or
+  exists(LogicalBinaryExpr lbe, DataFlow::Node lesser |
+    lbe.getAnOperand() = g.(DataFlow::RelationalComparisonNode).asExpr() and
+    e = lesser.asExpr() and
+    g.(DataFlow::RelationalComparisonNode).leq(branch, lesser, _, _)
+  )
```

Comment:
This disjunct is a subset of the first disjunct, and is therefore redundant. Did you have something else in mind?

Reply:
yes, you right, I'll remote it.